src/FileStorageQueue/fsq.h

```diff
@@ -413,7 +413,9 @@ class FSQ final : public CONFIG::T_FILE_NAMING_STRATEGY,
       if (next_file) {
         //      const FileProcessingResult result = processor_.OnFileReady(*next_file.get(),
         //      time_manager_.Now());
-        const bool successfully_processed = processor_.OnFileReady(next_file->full_path_name, next_file->size);
+        // AlexZ: we can't believe next_file->size here. Debugging shows that it can be greater than real file size.
+        // TODO: refactor FSQ or capture a bug.
```

Comment:
We do have to chat about it offline.

I'm happy to fix the bug, but a) do not yet understand which one are we talking about, b) the code has diverged quite a bit since then, and c) as I understand, you seem to have a functional workaround as of now.